api/response/schema.py

```diff
@@ -16,26 +17,39 @@ class Meta:
 
 class CreateResponse(graphene.Mutation):
     class Arguments:
-        question_id = graphene.Int(required=True)
+        question_id = graphene.List(graphene.Int, required=True)
         room_id = graphene.Int(required=True)
         rate = graphene.Int()
         text_area = graphene.String()
         missing_items = graphene.List(graphene.Int)
 
-    response = graphene.Field(Response)
+    response = graphene.List(Response)
 
     def mutate(self, info, **kwargs):
         validate_empty_fields(**kwargs)
         query = Room.get_query(info)
         room = query.filter_by(id=kwargs['room_id']).first()
         if not room:
             raise GraphQLError("Non-existent room id")
-        get_question = Question.query.filter_by(id=kwargs['question_id']).first() # noqa
-        if not get_question:
-            raise GraphQLError("Question does not exist")
-        question_type = get_question.question_type
-        resource = create_response(question_type, **kwargs)
-        return CreateResponse(response=resource)
+        question_ids = kwargs.pop('question_id')
+        resources = []
+        questions = Question.get_query(info).filter(
+            QuestionModel.id.in_(question_ids)).all()
+        valid_question_ids = set()
+        for question in questions:
+            valid_question_ids.add(question.id)
+            question_type = question.question_type
+            resource = create_response(
+                question_type, question_id=question.id, **kwargs)
+            resources.append(resource)
+        invalid_question_ids = set(question_ids).difference(
+            valid_question_ids)
+        if invalid_question_ids:
+            raise GraphQLError(
+                'Response for question ids {} were not saved'.format(
```

Comment:
Tell the user why they were not saved. Also, write the test for this.

Reply:
This has been implemented @Bonifase.